test/metric/container_insights.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"log"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+type ContainerInsightsValueFetcher struct {
+	baseMetricValueFetcher
+}
+
+var _ MetricValueFetcher = (*ContainerInsightsValueFetcher)(nil)
+
+func (f *ContainerInsightsValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
```

Comment:
**Nit**: Namespace should be unique for container insights (e.g AWS/ECS/ContainerInsights). Therefore, should we strive for a function call (determine Namespace based on EKS/ECS/EC2)

Reply:
I don't understand. Namespace should be unique? Unique in.. what? Also this entire fetch logic is changed here https://github.com/aws/amazon-cloudwatch-agent-test/pull/41 so let's talk over there

Reply:
> Therefore, should we strive for a function call

What does this mean? The test calls this and passes in `ECS/ContainerInsights`. That is sufficient.

Reply:
In Container Insights, there always a default namespace (e.g [EKS Container Insights](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/main/plugins/processors/k8sdecorator/structuredlogsadapter/metricruletagger.go#L13) and [ECS Container Insights](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/main/plugins/processors/ecsdecorator/ecsdecorator.go#L16-L18). Moreover, we have a function to detect if we run in [EC2, ECS, EKS](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/main/translator/translate/logs/metrics_collected/prometheus/emfprocessor/ruleMetricNamespace.go#L23-L39) so want to ask if we could use/improve this general function to detect the namespace beforehand.